core/core.go

```diff
@@ -731,8 +739,35 @@ func constructPeerHost(ctx context.Context, id peer.ID, ps pstore.Peerstore, bwr
 		hostOpts = append(hostOpts, p2pbhost.NATPortMap)
 	}
 
+	if !opts.DisableRelay {
+		filterRelayAddr := func(addrs []ma.Multiaddr) []ma.Multiaddr {
+			var raddrs []ma.Multiaddr
+			for _, addr := range addrs {
+				_, err := addr.ValueForProtocol(circuit.P_CIRCUIT)
+				if err == nil {
+					continue
+				}
+				raddrs = append(raddrs, addr)
+			}
+			return raddrs
+		}
+		hostOpts = append(hostOpts, p2pbhost.AddrsFactory(filterRelayAddr))
```

Comment:
Yeah, Since we're going to be using the address factory pretty soon anyways for limiting which addresses we announce: https://github.com/ipfs/go-ipfs/pull/3948